File: api/src/main/java/com/dkowalczyk/psinder_app/dto/UserDto.java
package com.dkowalczyk.psinder_app.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserDto {
    private Long id;
    private String name;
    private int age;
    private String bio;
    private DogDto dog;
    private String role;
}

File: mobile/app/api/types/userDto.ts
import { DogDto } from "./dogDto";

/**
 * 
 * @export
 * @interface UserDto
 */
export interface UserDto {
    /**
     * 
     * @type {number}
     * @memberof UserDto
     */
    'id'?: number;
    /**
     * 
     * @type {string}
     * @memberof UserDto
     */
    'name'?: string;
    /**
     * 
     * @type {number}
     * @memberof UserDto
     */
    'age'?: number;
    /**
     * 
     * @type {string}
     * @memberof UserDto
     */
    'bio'?: string;
    /**
     * 
     * @type {DogDto}
     * @memberof UserDto
     */
    'dog'?: DogDto;
    /**
     * 
     * @type {string}
     * @memberof UserDto
     */
    'role'?: string;
}